docs/draft/6-payloads.md

```diff
@@ -0,0 +1,359 @@
+  // so that we don't rely on it
+  uint64 timestamp = 2;
+  // Text of the message
+  string text = 3;
+  // Id of the message that we are replying to
+  string response_to = 4;
+  // Ens name of the sender
+  string ens_name = 5;
+  // Chat id, this field is symmetric for public-chats and private group chats,
+  // but asymmetric in case of one-to-ones, as the sender will use the chat-id
+  // of the received, while the receiver will use the chat-id of the sender.
+  // Probably should be the concatenation of sender-pk & receiver-pk in alphabetical order
+  string chat_id = 6;
+
+  // The type of message (public/one-to-one/private-group-chat)
+  MessageType message_type = 7;
+  // The type of the content of the message
+  ContentType content_type = 8;
+
+  oneof payload {
+    StickerMessage sticker = 9;
+    ImageMessage image = 10;
+  }
+
+  enum MessageType {
+    UNKNOWN_MESSAGE_TYPE = 0;
+    ONE_TO_ONE = 1;
+    PUBLIC_GROUP = 2;
+    PRIVATE_GROUP = 3;
+    // Only local
+    SYSTEM_MESSAGE_PRIVATE_GROUP = 4;}
+  enum ContentType {
+    UNKNOWN_CONTENT_TYPE = 0;
+    TEXT_PLAIN = 1;
+    STICKER = 2;
+    STATUS = 3;
+    EMOJI = 4;
+    TRANSACTION_COMMAND = 5;
+    // Only local
+    SYSTEM_MESSAGE_CONTENT_PRIVATE_GROUP = 6;
+    IMAGE = 7;
+  }
+}
+```
+
+#### Payload
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1 | clock | `uint64` | The clock of the chat|
+| 2 | timestamp | `uint64` | The sender timestamp at message creation |
+| 3 | text | `string` | The content of the message |
+| 4 | response_to | `string` | The ID of the message replied to |
+| 5 | ens_name | `string` | The ENS name of the user sending the message |
+| 6 | chat_id | `string` | The local ID of the chat the message is sent to |
+| 7 | message_type | `MessageType` | The type of message, different for one-to-one, public or group chats |
+| 8 | content_type | `ContentType` | The type of the content of the message | 
+| 9 | payload | `Sticker|Image|nil` | The payload of the message based on the content type |
+
+#### Content types
+
+Content types are required for a proper interpretation of incoming messages. Not each message is plain text but may carry a different information.
+
+The following content types MUST be supported:
+* `TEXT_PLAIN` identifies a message which content is a plaintext.
+
+There are also other content types that MAY be implemented by the client:
+* `STICKER`
+* `STATUS`
+* `EMOJI`
+* `TRANSACTION_COMMAND`
+* `IMAGE`
+
+##### Sticker content type
+
+A `ChatMessage` with `STICKER` `Content/Type` MUST also specify the ID of the `Pack` and 
+the `Hash` of the pack, in the `Sticker` field of `ChatMessage`
+
+```protobuf
+message StickerMessage {
+  string hash = 1;
+  int32 pack = 2;
+}
+```
+
+##### Image content type
+
+A `ChatMessage` with `IMAGE` `Content/Type` MUST also specify the `payload` of the image
+and the `type`
+
+```protobuf
+message ImageMessage {
+  bytes payload = 1;
```

Comment:
Does this assume an image fits into one message?

Reply:
Yes, 
if fits in one `protobuf` message.
The idea is that any multipart is not done at this level (i.e not at the application layer), but at a lower level, either before waku/encryption (that's to be decided). If the message is too big, the whole protobuf payload is chunked, so any application message that exceeds the maximum size can be chunked (although we haven't really talked about multipart as of now)